packages/client/wallets/smart-wallet/src/blockchain/wallets/account/passkey.ts

```diff
@@ -130,18 +137,35 @@ export class PasskeyCreationStrategy implements AccountCreationStrategy {
         return error;
     }
 
-    private decorate<Account extends KernelSmartAccount<EntryPoint>>(account: Account, passkeyName: string): Account {
+    private decorate<Account extends KernelSmartAccount<EntryPoint>>(
+        account: Account,
+        passkeyName: string,
+        walletParams: WalletParams
+    ): Account {
         return new Proxy(account, {
             get: (target, prop, receiver) => {
                 const original = Reflect.get(target, prop, receiver);
                 if (typeof original !== "function" || typeof prop !== "string" || !isAccountSigningMethod(prop)) {
                     return original;
                 }
 
+                const signer = walletParams.signer as PasskeySigner;
                 return async (...args: any[]) => {
+                    const isFirstTransaction = args?.[0]?.factoryData != null;
+                    if (
+                        isFirstTransaction &&
+                        prop === "signUserOperation" &&
+                        signer.type === "PASSKEY" &&
+                        signer.onFirstTimePasskeySigning != null
```

Comment:
this is a bit unwieldy for a single `if`–what do you think about breaking out into a helper var to better describe the scenario?
```
const isFirstPasskeySignatureForUserOp = isFirstTransaction && ...

Reply:
I tried this but then i get this TS error 😅 

![Screenshot 2024-09-25 at 8 47 23 AM](https://github.com/user-attachments/assets/48144fe0-cf8f-4351-8aa6-1e11d3a2238c)